tests/unit/parsers/test.manifestjson.js

```diff
@@ -3185,4 +3185,337 @@ describe('ManifestJSONParser', () => {
+              gecko: {
+                strict_min_version: '88.0',
+              },
+            },
+            permissions: [],
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'no permissions',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '88.0',
+              },
+            },
+            permissions: undefined,
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'multiple permissions but strict_min_version > min version',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '89.0',
+              },
+            },
+            permissions: ['bookmarks', 'alarms'],
+          },
+          restrictedPermissions: new Map([['alarms', '88.1.3']]),
+        },
+      ],
+      [
+        'no restricted permissions',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '87.0',
+              },
+            },
+            permissions: ['bookmarks', 'alarms'],
+          },
+          restrictedPermissions: new Map([]),
+        },
+      ],
+      [
+        'versions contain non-numeric chars',
+        {
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version: '1.1pre',
+              },
+            },
+            permissions: ['alarms'],
+          },
+          restrictedPermissions: new Map([['alarms', '1.1c']]),
+        },
+      ],
+    ])(
+      'does not report an error when: %s',
+      (title, { manifestProps, restrictedPermissions }) => {
+        const { linter } = validate({
+          manifestProps,
+          restrictedPermissions,
+        });
+        const { errors } = linter.collector;
+
+        // We cannot assert `parser.isValid` because some (unrelated) warnings
+        // could set this prop to `false`.
+        expect(errors.length).toEqual(0);
+      }
+    );
+
+    it.each(['1', 'a'])(
+      'reports an error when the strict_min_version value is invalid (%s)',
+      (strict_min_version) => {
+        const { parser, linter } = validate({
+          manifestProps: {
+            browser_specific_settings: {
+              gecko: {
+                strict_min_version,
+              },
+            },
+            permissions: ['alarms'],
+          },
+          restrictedPermissions: new Map([['alarms', '1.01']]),
+        });
+        const { errors } = linter.collector;
+
+        expect(parser.isValid).toEqual(false);
+        expect(errors.length).toEqual(2);
+        expect(errors[0].code).toEqual(messages.JSON_INVALID.code);
+        expect(errors[1].code).toEqual(messages.RESTRICTED_PERMISSION);
```

Comment:
As for the other assertion, I'd prefer this to use arrayContaing/objectContaining, and apparently we'll need to account for the other error to be MANIFEST_FIELD_INVALID when the strict_min_version property is not a string:

```diff
@@ -3491,9 +3497,19 @@ describe('ManifestJSONParser', () => {
         const { errors } = linter.collector;

         expect(parser.isValid).toEqual(false);
-        expect(errors.length).toEqual(2);
-        expect(errors[0].code).toEqual(messages.JSON_INVALID.code);
-        expect(errors[1].code).toEqual(messages.RESTRICTED_PERMISSION);
+        const errorOnInvalidStrictVersion = typeof strict_min_version == "string"
+          ? expect.objectContaining({ code: messages.JSON_INVALID.code })
+          : expect.objectContaining({ code: messages.MANIFEST_FIELD_INVALID.code })
+
+        expect(errors).toEqual(
+          expect.arrayContaining([
+            errorOnInvalidStrictVersion,
+            expect.objectContaining({
+              code: messages.RESTRICTED_PERMISSION,
+              file: 'manifest.json',
+            }),
+          ])
+        );
       }
     );
```